### PR TITLE
Fix linter

### PR DESCRIPTION
**Pull request by dvonessen**
What does this PR change? What problem does it solve?
-----------------------------------------------------

This PR fixes some issues with the linter. 
See https://gitlab.com/thomassteinbach/gitlab-ci-python-library/-/jobs/1161953981

Was the change discussed in an issue?
------------------------------------------------------------

None

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CONTRIBUTING.md)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [x] I have added api documentation for the changes
- [x] I added a changelog entry to the unreleased section with the PR number. [Changelog](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CHANGELOG.md)
- [x] Linting is done, and linting jobs are successfull
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 generate-pipeline:
   stage: build
-  image: python:3.9-slim
+  image: thomass/gcip
```
Comment (ThomasSteinbach) [inline .gitlab-ci.yml:4]: - I would like to not create a cyclic dependency here ("Python Project gcip" -> "Docker Image gcip" -> "Python Project gcip" -> ...)
- Even if we do, the way to use the Docker image looks like following:
  https://dbsystel.github.io/gitlab-ci-python-library/user/index.html#_configuring_your_project_to_use_gcip
- Using a "basic" python image prevents concerns about API breaks within our own project. The input (new gcip code) should not depend on the output (old gcip Docker image from old code).

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 * Normalize config_file_path in `gcip.addons.container.config.DockerClientConfig`
+* Line length check in flake8 linter to 160.
+* To generate the pipeline we are using thomass/gcip instead of vanilla python image.
```
Comment (ThomasSteinbach) [inline CHANGELOG.md:15]: If previous suggestion applied, then...

```suggestion
```

**Review by ThomasSteinbach — CHANGES_REQUESTED** [review]

**Comment by ThomasSteinbach** [comment]
@dvonessen you can merge yourself if you resolve both comments and if no further discussion is required

**Comment by dvonessen** [comment]
Reverted thomass/gcip -> python:3.9-slim

**Pull request merged**